test/topology_custom/test_raft_no_quorum.py

```diff
@@ -110,6 +110,55 @@ async def test_quorum_lost_during_node_join(manager: ManagerClient) -> None:
     await third_node_future
 
 
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_quorum_lost_during_node_join_response_handler(manager: ManagerClient) -> None:
+    logger.info("starting a first node (the leader)")
+    servers = [await manager.server_add()]
+
+    logger.info("starting a second node (the follower)")
+    servers += [await manager.server_add()]
+
+    logger.info("adding a third node")
+    servers += [await manager.server_add(config={
+        'error_injections_at_startup': [
+            {
+                'name': 'group0-raft-op-timeout-in-ms',
+                'value': '1000'
+            },
+            {
+                'name': 'raft-group-registry-fd-threshold-in-ms',
+                'value': '500'
+            },
+            {
+                'name': 'join-node-response_handler-before-read-barrier'
+            }
+        ]
+    }, start=False)]
+
+    logger.info("starting a third node")
+    third_node_future = asyncio.create_task(
+        manager.server_start(servers[2].server_id,
+                             expected_error="raft operation [read_barrier] timed out, no group0 quorum",
+                             timeout=60))
+
+    logger.info(f"waiting for the third node {servers[2]} to hit join-node-response_handler-before-read-barrier")
+    log_file = await manager.server_open_log(servers[2].server_id)
+    await log_file.wait_for("join-node-response_handler-before-read-barrier injection hit", timeout=60)
+
+    logger.info("stopping the second node")
+    await manager.server_stop_gracefully(servers[1].server_id)
+
+    logger.info("release join-node-response_handler-before-read-barrier injection")
+    injection_handler = InjectionHandler(manager.api,
+                                         'join-node-response_handler-before-read-barrier',
+                                         servers[2].ip_addr)
+    await injection_handler.message()
+
+    logger.info("waiting for third node joining process to fail")
+    await third_node_future
```

Comment:
Could we merge some of these test cases so not to start as many clusters/servers?

Reply:
I don't like squishing tests into one write-once-understand-never test because of infrastructure constraints. The tests are already quite hard to follow because of injections and asynchrony, it would be even harder if different scenarios started intertwining with each other.

It should be possible to ask the infra for a node and get back one quickly and cheaply. Wouldn't it be more efficient to have some 'reset' api in Scylla, which would drop all the data and recreate all the necessary services?

For now I'll think if I can merge some of the tests without introducing much complexity.